src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -79,50 +83,107 @@ public static void execute(String username, String password, String host, String
-		if (oldIdMappingFile.exists()) {
-			oldIdMappingFile.delete();
-		}
-		oldIdMappingFile.createNewFile();
-		String header = "# Reactome stable IDs for release " + releaseNumber + "\n" + "Stable_ID\told_identifier(s)\n";
-		Files.write(Paths.get(filename), header.getBytes(), StandardOpenOption.APPEND);
-		for (List<Object> stableIdsArray : preferredIds)
-		{
-			String primaryId = (String) stableIdsArray.get(0);
-			@SuppressWarnings("unchecked")
-			List<String> secondaryIds = (ArrayList<String>) stableIdsArray.get(1);
-			String line = primaryId + "\t" + String.join(",", secondaryIds) + "\n";
-			Files.write(Paths.get(filename), line.getBytes(), StandardOpenOption.APPEND);
-		}
-		String outpathName = releaseNumber + "/" + filename;
-		Files.move(Paths.get(filename), Paths.get(outpathName), StandardCopyOption.REPLACE_EXISTING);
+		Files.deleteIfExists(Paths.get(filename));
+		Files.createFile(Paths.get(filename));
+
+		writeMappingsToFile(releaseNumber, stableIdsToOldIdsMappings, currentStableIdentifiers);
 
 		logger.info("MapOldStableIds finished");
 	}
 
-	// Iterate through returned results of DB IDs and stable IDs
-	private static Map<String, List<String>> getDbIdToStableIds(ResultSet resultSet) throws SQLException {
+	/**
+	 * Creates a MySQL driver that queries the stable_identifiers database for identifier and instanceId from the StableIdentifier table.
+	 * @param dba MySQLAdaptor, Not used to query the database, but to supply database parameters like host, username and password.
+	 * @return ResultSet from MySQL stable_identifiers database that consists of identifiers and their associated instanceId
+	 * @throws ClassNotFoundException - Thrown if MySQL driver class isn't found
+	 * @throws SQLException - Thrown if there are issues connecting/querying/interacting with stable_identifiers database
+	 */
+	private static ResultSet retrieveAllStableIdentifiers(MySQLAdaptor dba) throws ClassNotFoundException, SQLException {
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifiers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		Connection connect = DriverManager.getConnection("jdbc:mysql://" + dba.getDBHost() + "/stable_identifiers?" + "user=" + dba.getDBUser() + "&password=" + dba.getDBPwd());
+		Statement statement = connect.createStatement();
+		return statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
+	}
+
+	/**
+	 * Checks that the primary identifier taken from the stable_identifiers database is currently used, and it has secondary mappings.
+	 * @param currentStableIdentifiers Set<String> - Set of all StableIdentifiers currently in database.
+	 * @param primaryId String - Primary StableIdentifier that maps to secondaryIds.
+	 * @param secondaryIds List<String> - All StableIdentifiers (old and new formats) that map to the primary stable identifier.
+	 * @return Boolean, indicating it is a currently used StableIdentifier with secondary mappings.
```

Comment:
Nit-pick :) - Boolean should be lower case to indicate a primitive type is returned.

Reply:
But... it's the beginning of the sentence, so a capital letter is the correct way to start a sentence in English. And since there is no markup to indicate that "Boolean" is a keyword (and Javadoc allows markup for the purpose), this could be correct.

A good compromise would be to change the beginning of the sentence to "A boolean..." ;)